eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ContextualClientProvider.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Generator.Management.Models;
+using Azure.Generator.Management.Snippets;
+using Azure.Generator.Management.Utilities;
+using Azure.Generator.Management.Visitors;
+using Azure.ResourceManager;
+using Microsoft.TypeSpec.Generator.Expressions;
+using Microsoft.TypeSpec.Generator.Input;
+using Microsoft.TypeSpec.Generator.Input.Extensions;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Snippets;
+using Microsoft.TypeSpec.Generator.Statements;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using static Microsoft.TypeSpec.Generator.Snippets.Snippet;
+
+namespace Azure.Generator.Management.Providers
+{
```

Comment:
Given this class doesn't really have any functionality as a TypeProvider, it takes in contextualRequestPattern and populate arguments.
Instead of letting RsourceClientProvdier and ResourceCollectionClientProvider inherit this, how about make it as composition within them as a utility?

Reply:
We already do it as a composition right now - the contextual parameters are in a dictionary or an array, we are only one step away from that (for instance putting them inside a class to qeury). This is fine.
If we do not make this as the base type, in those XXXMethodProvider methods, we will have to pass in this `ContextualParameterProvider` into everything as a verbose version of `RequestPathPattern`.
Would this be OK?

Reply:
I think so, or instantiate ContextualParameterProvider in ResourceOperationMethodProvider directly.
related to https://github.com/Azure/azure-sdk-for-net/pull/51169/files#r2199695803